core/src/main/java/io/questdb/std/Unsafe.java

```diff
@@ -277,7 +277,10 @@ public static long malloc(long size, int memoryTag) {
     public static long realloc(long address, long oldSize, long newSize, int memoryTag) {
         try {
             assert memoryTag >= MemoryTag.NATIVE_PATH;
-            checkAllocLimit(-oldSize + newSize, memoryTag);
+            // Check the limit unless realloc is going to free some memory.
+            if (newSize >= oldSize) {
```

Comment:
With `newSize <= oldSize`, `checkAllocLimit` returns immediately. This seems to be a duplicate check for the same.